jbmc/src/java_bytecode/java_bytecode_parser.cpp

```diff
@@ -444,28 +444,37 @@ bool java_bytecode_parsert::parse()
   return false;
 }
 
-#define ACC_PUBLIC       0x0001
-#define ACC_PRIVATE      0x0002
-#define ACC_PROTECTED    0x0004
-#define ACC_STATIC       0x0008
-#define ACC_FINAL        0x0010
-#define ACC_SYNCHRONIZED 0x0020
-#define ACC_BRIDGE       0x0040
-#define ACC_VARARGS      0x0080
-#define ACC_NATIVE       0x0100
-#define ACC_INTERFACE    0x0200
-#define ACC_ABSTRACT     0x0400
-#define ACC_STRICT       0x0800
-#define ACC_SYNTHETIC    0x1000
-#define ACC_ANNOTATION   0x2000
-#define ACC_ENUM         0x4000
-
 #ifdef _MSC_VER
 #define UNUSED
 #else
 #define UNUSED __attribute__((unused))
 #endif
 
+/// Represents the bit flag from class access and property flags
+/// https://docs.oracle.com/javase/specs/jvms/se8/html/jvms-4.html#jvms-4.1-200-E.1
+namespace class_access_flagst
```

Comment:
Using `enum class` should remove this weird namespace.

Reply:
Unfortunately using `enum class` means it cannot be implicitly converted to `int` (to be used in the logical and later). I _could_ replace the word `namespace` with `class`, but that feels weirder.

Reply:
How about doing away with the defines / enums and using this instead?
```
  struct flagst {
    bool is_public : 1;
    bool is_private : 1;
    bool is_protected : 1;
    bool is_static : 1;
    bool is_final : 1;
    void : 4;
    bool is_interface : 1;
    bool is_abstract : 1;
    void : 1;
    bool is_synthetic : 1;
    bool is_annotation : 1;
    bool is_enum : 1;
    void : 1;
  } flags = *reinterpret_cast<flagst*>(&access_flags);

  parsed_class.is_abstract = flags.is_abstract;
  parsed_class.is_enum = flags.is_enum; 
  parsed_class.is_final = flags.is_final; 
  parsed_class.is_interface = flags.is_interface;
  parsed_class.is_synthetic = flags.is_synthetic;
  parsed_class.is_annotation = flags.is_annotation;
```

Reply:
@peterschrammel I really like Tom's suggestion above, if you'd like it I'll go ahead and change it for that with some unit tests to check we get each method correct.

Reply:
@peterschrammel Going to pull this refactor out into a different PR to unblock this - but could you express an opinion on whether you like Toms suggested approach (I think I'd probably put the struct directly into `parse_treet::classt` to avoid the `is_abstract=is_abstract` duplication.